openfe_gromacs/protocols/gromacs_md/md_methods.py

```diff
@@ -654,3 +848,228 @@ def _execute(
+        cpt,
+        log,
+        edr,
+        shared_basebath,
+    ):
+        assert os.path.exists(in_gro)
+        assert os.path.exists(top)
+        assert os.path.exists(mdp)
+        p = subprocess.Popen(
+            [
+                "gmx",
+                "grompp",
+                "-f",
+                mdp,
+                "-c",
+                in_gro,
+                "-p",
+                top,
+                "-o",
+                tpr,
+            ],
+            stdin=subprocess.PIPE,
+        )
+        p.wait()
+        assert os.path.exists(tpr)
+        p = subprocess.Popen(
+            [
+                "gmx",
+                "mdrun",
+                "-s",
+                tpr.name,
+                "-cpo",
+                cpt,
+                "-o",
+                trr,
+                "-x",
+                xtc,
+                "-c",
+                out_gro,
+                "-e",
+                edr,
+                "-g",
+                log,
+            ],
+            stdin=subprocess.PIPE,
+            cwd=shared_basebath,
+        )
+        p.wait()
+        return
+
+    def _execute(
+        self,
+        ctx: gufe.Context,
+        *,
+        protocol,
+        setup,
+        verbose=True,
+        **kwargs,
+    ) -> dict[str, Any]:
+        """
+        Execute the simulation part of the Gromacs MD protocol.
+
+        Parameters
+        ----------
+        ctx : gufe.protocols.protocolunit.Context
+            The gufe context for the unit.
+        protocol : gufe.protocols.Protocol
+            The Protocol used to create this Unit. Contains key
+            information
+            such as the settings.
+        setup : gufe.protocols.ProtocolUnit
+            The SetupUnit
+
+        Returns
+        -------
+        dict : dict[str, str]
+            Dictionary with paths to ...
+        """
+        log_system_probe(logging.INFO, paths=[ctx.scratch])
+
+        if ctx.shared is None:
+            # use cwd
+            shared_basepath = pathlib.Path(".")
+        else:
+            shared_basepath = ctx.shared
+
+        # ToDo: Figure out how to specify the order in which to run things
+        # ToDo: Add output settings, e.g. name of output files
+        protocol_settings: GromacsMDProtocolSettings = self._inputs["protocol"].settings
+        sim_settings_em: EMSimulationSettings = protocol_settings.simulation_settings_em
+        sim_settings_nvt: NVTSimulationSettings = (
+            protocol_settings.simulation_settings_nvt
+        )
+        sim_settings_npt: NPTSimulationSettings = (
+            protocol_settings.simulation_settings_npt
+        )
+        output_settings_em: EMOutputSettings = protocol_settings.output_settings_em
+        output_settings_nvt: NVTOutputSettings = protocol_settings.output_settings_nvt
+        output_settings_npt: NPTOutputSettings = protocol_settings.output_settings_npt
+
+        input_gro = setup.outputs["system_gro"]
```

Comment:
TODO - need to make it pick up "previous step" gro file.

Reply:
Done